tests/unit/DiffInvalidationEvent.cpp

```diff
@@ -0,0 +1,39 @@
+/// \file DiffInvalidationEvent.cpp
+/// \brief Tests for C API of revng-pipeline
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipes/Kinds.h"
+#include "revng/Pipes/ModelInvalidationEvent.h"
+#include "revng/Support/MetaAddress.h"
+
+#define BOOST_TEST_MODULE PipelineC
+bool init_unit_test();
+#include "boost/test/unit_test.hpp"
+
+#include "revng/UnitTestHelpers/UnitTestHelpers.h"
+
+using namespace revng::pipes;
+using namespace model;
+using namespace pipeline;
+
+BOOST_AUTO_TEST_SUITE(ModelInvalidationDiffSuite)
+
+BOOST_AUTO_TEST_CASE(RootInvalidationTest) {
+  model::Binary Empty;
+  model::Binary New;
+
+  MetaAddress Address(0x1000, MetaAddressType::Code_aarch64);
+  New.ExtraCodeAddresses.insert(Address);
+
+  auto Diff = diff(Empty, New);
+
+  TargetsList ToRemove;
+  Root.getInvalidations(ToRemove, Diff);
+  BOOST_TEST(ToRemove.size() == 1);
+  BOOST_TEST(&ToRemove.front().getKind() == &Root);
```

Comment:
Oh gosh. This kinda nice, but it's also weird. So we produce the list of *all* the targets that need to be invalidated, no matter whether they are actually present?
It's elegant but I'm a bit concerned about performance, aren't we wasting computation for stuff that's not there in the first place?

Probably not the bottleneck, but give it a think.

Reply:
it was like you say before, but we decided to have it this way so that it can be tested without end to end testing infrastructure

Reply:
OK, hopefully this won't become a bottleneck.